### DISCUSSION TITLE
Typing text + reverse + loop

### DISCUSSION THREAD

**Discussion by Ionic** [exeldro/obs-move-transition#239] [Q&A] [answered]
I'd like to animate some text, specifically three dots.

The animation is supposed to blend in each dot individually, then blend them out and restart.

Visual representation: "" -> "." -> ".." -> "..." -> ".." -> "." -> "" (and back to the beginning)

Two iterations of that work fine by using a filter (Typeout 1) that fires directly with "Next Move" set to "Reverse" and a simultaneous move (Typeout 2) that is a copy of Typeout 1 with Start Delay set to 10s, "Next Move" set to "Reverse" and a simultaneous move to Typeout 3.

Here, things go haywire.

I can give Typeout 3 a start delay of, say, 20s, but I'm unable to create an actual loop. I'd have to set the simultaneous move of Typeout 3 to Typeout 2, but that doesn't create an infinite loop, since the simultaneous move will not trigger after the start delay, but directly at the start. Unfortunately, triggering Typeout 2 at that point is useless, since it's already running.

This is pretty annoying. The central issue is that "Reverse" is a "Next Move" action, which means that you essentially can't chain a move value filter in a meaningful way when using the reverse feature. Using "Simultaneous Move" as a workaround doesn't really work either, because it's firing right at the start and filters can't be triggered if they are already running.

Am I missing something here?

**Comment by exeldro** [comment]
Instead of reverse you can use an other move value filter to go back

**Comment by Ionic** [comment]
If it was a normal move operation, that would be true, yes.

For a "typing" filter, I can only really emulate the reverse action with n additional filters that remove one character per filter. A workaround that I can employ, sure, especially for three characters only, but it quickly gets unwieldy for longer strings.

I still believe that "reverse" should be an internal property of the filter and not a "next move" target. I might try to implement this once I can split off some time.

**Comment by exeldro** [comment, marked as answer]
Why would you need a filter per character? A move value filter can remove multiple characters.

**Reply by Ionic** [reply]
Oh, you're absolutely right! Clearing the text and setting the animation length to a suitable value clears out every character one by one. That's easily loopable! Thanks!
